### ISSUE TITLE
Problem with CDB list

### ISSUE THREAD

**Issue by goldarai** [wazuh/wazuh#4136]
| 3.10.2 | - | Manager | Packages | Ubuntu 18.04 |
Wazuh Agents deploy: +10K | Wazuh managers: 10

Hi Wazuh team.

We have problems with a CDB list. It looks similar the following issue: #4067  

Our goal is alert all interactive commands (tty!=none) performed by service accounts (0<auid<1100 and auid!=4294967295). To do that we have three custom rules and one CDB list. Sometimes we got an alert from a IP in the list, and that shouldn't be happend.

```
    <rule id="106000" level="9">
         <!-- Default rule - Audit:  Command -->
         <if_sid>80792</if_sid>
         <field name="audit.tty">\.+</field>
         <description>Interactive command with associated tty.</description>
    </rule>
```

```
<rule id="107062" level="2">
    <if_sid>106000</if_sid>
    <list field="audit.auid" lookup="not_match_key">etc/lists/allowedInteractiveauids</list>
    <description>Interactive command with daemon account</description>
</rule>
```

```
<rule id="107063" level="11">
        <if_sid>107062</if_sid>	
        <description>Interactive command with daemon auid (hacking in progress!!)</description>	        
 </rule>
```

Extract of file allowedInteractiveauids
```
0:
1000:
1001:
1002:
1003:
1004:
1005:
...
1099:
4294967295:
```

If we use the field filter (instead of using a CDB list) we don't have problems.
```
    <rule id="107062" level="2">
        <if_sid>106000</if_sid>
        <description>Interactive command with daemon auid (hacking in progress, transitional alert)</description>
    </rule>
```

```
    <rule id="107063" level="11">
        <if_sid>107062</if_sid>
        <description>Interactive command with daemon auid (hacking in progress!!)</description>
    </rule>
``` 

```
    <rule id="106013" level="7">
        <if_sid>107063</if_sid>
        <field name="audit.auid">0|1001|1002|1003|1004|1005|1006|1007|1008|1009|1010|1011|1012|1013|1014|1015|1016|1017|1018|1019|1020|1021|1022|1023|1024|1025|1026|1027|1028|1029|1030|1031|1032|1033|1034|1035|1036|1037|1038|1039|1040|1041|1042|1043|1044|1045|1046|1047|1048|1049|1050|1051|1052|1053|1054|1055|1056|1057|1058|1059|1060|1061|1062|1063|1064|1065|1066|1067|1068|1069|1070|1071|1072|1073|1074|1075|1076|1077|1078|1079|1080|1081|1082|1083|1084|1085|1086|1087|1088|1089|1090|1091|1092|1093|1094|1095|1096|1097|1098|1099|4294967295</field>
        <description>Interactive command with associated tty</description>
    </rule>
```

Could someone give us a hand with this issue? 

Regards.

**Comment by Zenidd** [comment]
Hello @goldarai,
 
Let's assure that the configuration is the right one:
- You should check if the CDB list is found in every worker even when the master node should provide the CDB lists centralization.
- The master doesn’t send its local configuration file to the workers. If the configuration is changed in the master node, it should be changed manually in the workers. Please, check if all your workers have the related `<list>etc/lists/your_cbd_list</list>` entry in its `/var/ossec/etc/ossec.conf` file
- The workers should be restarted after the CDB list synchronization and the ossec.conf editing. This restart does not occur automatically.

To summarize, each agent should have the CDB list, the `<list>etc/lists/your_cbd_list</list>` `ossec.conf` entry and should be restarted when the two previous requisites are ready.

Greetings, JP Sáez

**Comment by goldarai** [comment]
Hi @Zenidd 

- The CDB list is in every worker and the master.
- All the workers and the master have the <list>etc/lists/allowedInteractiveauids</list> in its ossec.conf
- We restarted all the workers and the master after add this list.
- We checked the file permissions and it is OK.
rw-rw---- 1 ossec ossec  615 Oct 21 20:10 allowedInteractiveauids

Regards.

**Comment by Zenidd** [comment]
Hello @goldarai,

First of all thanks for your detailed reply. Everything seems ok so I think that you could paste here some false positive alert so that way we can work around something.

We can extract useful information and see if these false positives have always the same manager node source or other correlations that lead us to solve this.

(Feel free to obfuscate any sensitive data)

Greetings, JP Sáez

**Comment by goldarai** [comment]
Hi @Zenidd 

For example the following alert was fired, but the auid 4294967295 is in the list.

{
  "_index": "linux-critical-2019.10.22",
  "_type": "_doc",
  "_id": "43DJ8G0B-V5EGPYL4-0s",
  "_version": 1,
  "_score": null,
  "_source": {
    "ecs": {
      "version": "1.0.1"
    },
    "agent": {
      "ephemeral_id": "c73740a9-bb1c-4d4a-ae11-b268eb5d0e63",
      "version": "7.3.2",
      "name": "bot-xxxx-2-08",
      "id": "f723d7af-4a3b-4d8a-9d47-0424dffc4954",
      "ip": "10.172.xxx.xxx",
      "type": "filebeat",
      "hostname": "xxx-wazuh-mg-14"
    },
    "rule": {
      "mail": false,
      "description": "Interactive command with daemon auid (hacking in progress!!)",
      "firedtimes": 1040,
      "level": 11,
      "id": 107063,
      "groups": [
        "audit",
        "windows"
      ]
    },
    "redirect": "linux-critical",
    "input": {
      "type": "log"
    },
    "command": "awk -v mimetype=x-scheme-handler/https A20202020424547494E.... /home/xxxxxxx/.config/mimeapps.list ",
    "location": "/var/log/audit/audit.log",
    "@timestamp": "2019-10-22T00:07:03.878Z",
    "manager": {
      "name": "xxx-wazuh-mg-14"
    },
    "full_log": "type=SYSCALL msg=audit(1571702786.905:1272939): arch=c000003e syscall=59 success=yes exit=0 a0=55ce2cf9c5d8 a1=55ce2cf9c490 a2=55ce2cf9c4c0 a3=1 items=2 ppid=17018 pid=17027 auid=4294967295 uid=1001 gid=1001 euid=1001 suid=1001 fsuid=1001 egid=1001 sgid=1001 fsgid=1001 tty=pts10 ses=4294967295 comm=\"awk\" exe=\"/usr/bin/gawk\" key=\"audit-wazuh-c\" type=EXECVE msg=audit(1571702786.905:1272939): argc=5 a0=\"awk\" a1=\"-v\" a2=\"mimetype=x-scheme-handler/https\" a3=0A20202020424547494E... a4=\"/home/xxxxxxx/.config/mimeapps.list\" type=CWD msg=audit(1571702786.905:1272939):  cwd=\"/home/xxxxxxx\" type=PATH msg=audit(1571702786.905:1272939): item=0 name=\"/usr/bin/awk\" inode=24332 dev=fd:01 mode=0100755 ouid=0 ogid=0 rdev=00:00 nametype=NORMAL type=PATH msg=audit(1571702786.905:1272939): item=1 name=\"/lib64/ld-linux-x86-64.so.2\" inode=28218 dev=fd:01 mode=0100755 ouid=0 ogid=0 rdev=00:00 nametype=NORMAL type=PROCTITLE msg=audit(1571702786.905:1272939): proctitle=61776B002D...",
    "decoder": {
      "name": "auditd",
      "parent": "auditd"
    },
    "id": "1571702823.135476826",
    "cluster": {
      "name": "xxx_wazuh_mg",
      "node": "xxx-wazuh-mg-14"
    },
    "data": {
      "audit": {
        "uid": "1001",
        "cwd": "/home/xxxxxxx",
        "auid": "4294967295",
        "command": "awk",
        "egid": "1001",
        "pid": "17027",
        "syscall": "59",
        "id": "1272939",
        "sgid": "1001",
        "type": "SYSCALL",
        "execve": {
          "a0": "awk",
          "a2": "mimetype=x-scheme-handler/https",
          "a1": "-v",
          "a4": "/home/xxxxxxx/.config/mimeapps.list"
        },
        "fsgid": "1001",
        "exit": "0",
        "euid": "1001",
        "fsuid": "1001",
        "arch": "c000003e",
        "success": "yes",
        "ppid": "17018",
        "tty": "pts10",
        "session": "4294967295",
        "suid": "1001",
        "key": "audit-wazuh-c",
        "file": {
          "name": "/usr/bin/awk",
          "mode": "0100755",
          "inode": "24332"
        },
        "exe": "/usr/bin/gawk",
        "gid": "1001"
      }
    }
  },
  "fields": {
    "@timestamp": [
      "2019-10-22T00:07:03.878Z"
    ]
  },
  "sort": [
    1571702823878
  ]
}

Regards.

**Comment by Zenidd** [comment]
Hello @goldarai,

First of all, sorry for the late reply. We have been investigating and recreating the environment.

We think that the problem lies in incorrect handling when a large number of events have to be checked against CDB lists. Anyway, we need to perform more tests to really know why is this happening.

I'm closing this issue and labeling it as duplicate: the bug is documented in an older issue. You can track the fixing progress there and add any information you consider useful. I would like to know if the filtering fail keeps happening in your environment.

I also want you to know that you can contact us through our [#community Slack channel](https://wazuh.com/community/join-us-on-slack/) and our [mailing list](https://groups.google.com/forum/?hl=es#!forum/wazuh).

Greetings, JP Sáez
